src/services/data-importer/csv/CSVDataImporter.ts

```diff
@@ -18,60 +18,54 @@
 
 import fs from 'fs-extra';
 import papaparse from 'papaparse';
+import { ObjectSchemaProperty, PropertyType } from 'realm';
 
-import { DataImporter } from '../DataImporter';
+import { DataImporter, ImportableFile } from '../DataImporter';
 
 export class CSVDataImporter extends DataImporter {
   private static readonly NUMBER_OF_INSERTS_BEFORE_COMMIT = 10000;
```

Comment:
(I know this isn't a part of the changes, but...) Does batch commits make sense in transactions? I would argue it contradicts transactions.

Reply:
It makes sense? The implementation opens a write transaction, inserts up to 10000 objects and then commits and opens a new transaction. It's like wrapping every 10000 `create` calls in a `write` call.

Reply:
Yes, but if you are importing a 20.000 rows csv, would you be interested in just the first 10.000 in the db, if something fails for the remaining 10.000? (Isn't that what potentially could happen?)

Reply:
You're right 👍  I think there's a trade-off here, but I can't remember why we shouldn't just make one huge transaction.